database/services/save.go

```diff
@@ -6,16 +6,17 @@ import (
 )
 
 // Save stores a service in the database and returns a hash or an error.
-func Save(service *service.Service) (hash string, err error) {
+func Save(service *service.Service) error {
 	bytes, err := proto.Marshal(service)
 	if err != nil {
-		return "", err
+		return err
 	}
 	db, err := open()
 	defer close()
 	if err != nil {
-		return "", err
+		return err
 	}
-	hash = service.Hash()
-	return hash, db.Put([]byte(hash), bytes, nil)
+	hash := service.Hash()
+	service.Id = hash
```

Comment:
I made this change because I needed to return _*service.Service_ type in [deploy api](https://github.com/mesg-foundation/core/blob/5e7fd59d0394b9a0cb387df96261f43ac1bd1850/mesg/deploy.go#L21). And its callers needs to read service's ID. 

I know this is messy but it is a necessary evil during this big refactor :) If you really don't like it I can remove it from this PR and make changes on deploy api to cover it but eventually deploy api  should return a proper Service type with id. Also if you review the corresponding commits related with this you can see all the changes made to _service.ID_ in one place. Easier to review.

Reply:
After second check, I see that I didn't separate changes for service.Id to different commits. fyi.